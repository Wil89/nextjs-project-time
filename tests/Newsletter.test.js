import { cleanup, render, fireEvent } from "@testing-library/react";
import NewsletterRegistration from "../components/input/newsletter-registration";
import "@testing-library/jest-dom";
import Enzyme, {mount} from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({adapter: new Adapter() });

describe("newsletter registration tests", () => {
  afterEach(cleanup);

  test("should match snapshot", () => {
    const { container } = render(<NewsletterRegistration />);
    expect(container).toMatchSnapshot();
  });

  test("should fire registration event", async () => {
    const handlerSubmitMock = jest.fn();
    const wrapper = mount(<NewsletterRegistration handlerRegistration={handlerSubmitMock} />)
    wrapper.find('input').simulate('change', {target: {value: 'wuj@gmail.com'}});
    wrapper.find('form').simulate('submit');
    expect(handlerSubmitMock).toBeCalled();
  });
});
