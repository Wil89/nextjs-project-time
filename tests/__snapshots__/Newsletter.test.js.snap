// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newsletter registration tests should match snapshot 1`] = `
<div>
  <div
    class="newsletter-registration__NewsletterRegistration-sc-wqc6tn-1 eKtYc"
  >
    <section
      class="newsletter-registration__Section-sc-wqc6tn-0 eIyvMo"
    >
      <h2>
        Sign up to stay updated!
      </h2>
      <form>
        <div>
          <input
            aria-label="Your email"
            data-testid="data-input"
            id="email"
            name="email"
            placeholder="Your email"
            type="email"
            value=""
          />
          <button>
            Register
          </button>
        </div>
      </form>
    </section>
  </div>
</div>
`;
